File: app/src/main/java/org/wikipedia/feed/dataclient/FeedClient.java
package org.wikipedia.feed.dataclient;

import android.content.Context;
import android.support.annotation.NonNull;

import org.wikipedia.dataclient.WikiSite;
import org.wikipedia.feed.model.Card;

import java.util.List;

public interface FeedClient {
    void request(@NonNull Context context, @NonNull WikiSite wiki, int age, @NonNull final Callback cb);
    void cancel();

    interface Callback {
        void success(@NonNull List<? extends Card> cards);
        void error(@NonNull Throwable caught);
    }
}

File: app/src/main/java/org/wikipedia/feed/dataclient/FeedClient.kt
package org.wikipedia.feed.dataclient

import android.content.Context
import org.wikipedia.dataclient.WikiSite
import org.wikipedia.feed.model.Card

interface FeedClient {

    interface Callback {
        fun success(cards: List<Card>)
        fun error(caught: Throwable)
    }

    fun request(context: Context, wiki: WikiSite, age: Int, cb: Callback)
    fun cancel()
}
